src/openassetio-core/src/utils/formatter.hpp

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <memory>
+#include <ostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <fmt/format.h>
+#include <openassetio/managerApi/ManagerInterface.hpp>
+#include <openassetio/trait/collection.hpp>
+#include <openassetio/typedefs.hpp>
+
+OPENASSETIO_FWD_DECLARE_IN_NS(errors, BatchElementError)
+OPENASSETIO_FWD_DECLARE(EntityReference)
+OPENASSETIO_FWD_DECLARE_TOP_LEVEL_NS(Context)
+OPENASSETIO_FWD_DECLARE_IN_NS(hostApi, Manager)
+OPENASSETIO_FWD_DECLARE_IN_NS(managerApi, HostSession)
+OPENASSETIO_FWD_DECLARE_IN_NS(trait, TraitsData)
+
+// Provide fmt formatters for most OpenAssetIO types.
+// Whilst this is internal, it is implementation for public behaviour, such
+// as << operators and python `str` and `repr`
+
+template <>
+struct fmt::formatter<openassetio::EntityReference> : fmt::formatter<string_view> {
+  auto format(const openassetio::EntityReference& entityRef, format_context& ctx) const
+      -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::EntityReferences> : fmt::formatter<string_view> {
+  auto format(const openassetio::EntityReferences& entityRefs, format_context& ctx) const
+      -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::StrMap> : fmt::formatter<string_view> {
+  auto format(const openassetio::StrMap& strMap, format_context& ctx) const -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::managerApi::ManagerInterface::Capability>
+    : fmt::formatter<string_view> {
+  auto format(openassetio::managerApi::ManagerInterface::Capability capability,
+              format_context& ctx) const -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::errors::BatchElementError::ErrorCode>
+    : fmt::formatter<string_view> {
+  auto format(openassetio::errors::BatchElementError::ErrorCode errorCode,
+              format_context& ctx) const -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::errors::BatchElementError> : fmt::formatter<string_view> {
+  auto format(const openassetio::errors::BatchElementError& ber, format_context& ctx) const
+      -> decltype(ctx.out());
+};
+
+template <>
+struct fmt::formatter<openassetio::trait::TraitSet> : fmt::formatter<string_view> {
```

Comment:
fmt has its own implementation of `string_view. `basic_string_view`, which this aliases to.

The docs : 
```
`fmt::basic_string_view`` is used for format strings even if ``std::string_view`` is available to prevent issues when a library is compiled with a different ``-std`` option than the client code (which is not recommended).
```

As for your second question, no I didn't. I figured we're trying to make strings, we should just use strings. There's more than enough trying to be clever going on as it is.